akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/javadsl/ClusterSharding.scala

```diff
@@ -365,18 +296,30 @@ final class Entity[M, E] private (
 }
 
 /**
- * Parameter to [[Entity.of]]
+ * Parameter to `createBehavior` function in [[Entity.of]].
+ *
+ * Cluster Sharding is often used together with [[akka.persistence.typed.javadsl.EventSourcedBehavior]]
+ * for the entities. To make that combination convenient to use the `EntityContext` has a `persistenceIdProposal`
+ * which can be used as the [[PersistenceId]] of the `EventSourcedBehavior`. It is constructed from the `name`
+ * of the [[EntityTypeKey]] and the `entityId` by concatenating them with `|` separator, or the separator that was
+ * defined by [[EntityTypeKey#withEntityIdSeparator]]. It's not mandatory to use exactly this `PersistenceId`.
+ * If `EventSourcedBehavior` isn't used the `persistenceIdProposal` can be ignored.
+ *
+ * @param entityId the business domain identifier of the entity
+ * @param persistenceIdProposal suggestion of [[PersistenceId]] to be used with an
+ *                              [[akka.persistence.typed.javadsl.EventSourcedBehavior]]
  */
 final class EntityContext[M](
     entityId: String,
-    shard: ActorRef[ClusterSharding.ShardCommand],
-    actorContext: ActorContext[M]) {
+    persistenceIdProposal: PersistenceId,
+    shard: ActorRef[ClusterSharding.ShardCommand]) {
 
   def getEntityId: String = entityId
 
+  def getPersistenceIdProposal: PersistenceId = persistenceIdProposal
+
   def getShard: ActorRef[ClusterSharding.ShardCommand] = shard
```

Comment:
(related to https://github.com/akka/akka/pull/27725#discussion_r325515363)

I assume the purpose is to bring Persitence API and Sharding API closer. I wonder if these helpers `EntityContext => PersistenceId` should be part of Sharding or, instead, they should be part of Persistence. I have the gut feeling that it's Persistence that needs Sharding and not the other way around.

Similarly, Projections need Sharding and I think it'd be a mistake if we polluted `EntityContext` with `EntityContext => WorkerCoordinates` (ref [`WorkerCoordinates`](https://github.com/lagom/lagom/blob/master/projection/core/src/main/scala/com/lightbend/lagom/internal/projection/ProjectionRegistryActor.scala#L47) ).